### DISCUSSION TITLE
Should I use Objects for unique structs?

### DISCUSSION THREAD

**Discussion by 0xmaayan** [aptos-labs/aptos-developer-discussions#307] [Questions] [answered]
### Discord user ID

Enjin

### Describe your question in detail.

Hello, I had a question for how/when to use objects. I have a list of structs that I want to keep unique, and I'm curious if i can use an id field to identify them, or if its best to make them each an object. I would also need a way to identify each struct (i.e. being able to pass it in as parameter) 

### Which operating system are you using?

N/A

### Which SDK or tool are you using? (if any)

N/A

**Comment by MoonShiesty** [comment]
objects come with a  unique id (address) so the object approach makes sense to me

you can pass and store global references to objects with Object\<MyStruct\>

**Reply by RevanthGundala** [reply]
When would you not want to use an object? My understanding is that objects only make sense in a scenario where you want ownership of an asset, otherwise it's better to just use an arbitrary id field.

**Reply by MoonShiesty** [reply, marked as answer]
i wouldn't use an object for singletons that don't have an owner, for example a GlobalConfig struct

objects allow you to type check, store and pass global references through Object<T>

objects also allow you manage lifetimes, you can delete/transfer/upgrade through refs

aptos also supports named objects, which have a deterministic mapping between some key (name) and object address

if you really don't need those features an id probably works too
